Align mathvariant attribute with MathML Core

eerii opened the pull request:
#### 5639c8e5efb63017fc4bf1b91d274b6b5a4a0dc5
<pre>
Align mathvariant attribute with MathML Core
<a href="https://bugs.webkit.org/show_bug.cgi?id=202302">https://bugs.webkit.org/show_bug.cgi?id=202302</a>

Reviewed by Frédéric Wang Nélar.

The CoreMathMLDeprecateLegacyMathvariant preference is used to disable the
previous mathvariant behaviour and other variations. The UA stylesheet is
updated to align with the spec, implementing the italic transform on mi
nodes using text-transform: math-auto.

The mathvariant attribute can only be present on mi nodes
(<a href="https://w3c.github.io/mathml-core/#dfn-mathvariant).">https://w3c.github.io/mathml-core/#dfn-mathvariant).</a> If specified,
it resets the text-transform to none using a presentational hint.

* LayoutTests/accessibility/ios-simulator/math.html:
* LayoutTests/accessibility/mac/math-alttext.html:
* LayoutTests/accessibility/mac/mathml-elements.html:
* LayoutTests/accessibility/mac/mathml-root.html:
* LayoutTests/accessibility/math-fenced.html:
* LayoutTests/accessibility/math-mo-fences-and-separators.html:
* LayoutTests/accessibility/math-multiscript-attributes.html:
* LayoutTests/accessibility/math-text.html:
* LayoutTests/fast/css/readonly-pseudoclass-common-element-expected.txt:
* LayoutTests/fast/css/readwrite-pseudoclass-editable-expected.txt:
* LayoutTests/fast/layers/accumulated-offset-overflow-Render-geometry-map-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/mathml-html-serializtion-on-copy.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/documents/dom-tree-accessors/document.getElementsByName/document.getElementsByName-namespace-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/documents/dom-tree-accessors/document.getElementsByName/document.getElementsByName-namespace-xhtml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/math-parse03-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/direction/direction-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/size-and-position-of-stretchy-fences-with-default-font-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/attribute-mapping-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/floats/not-floating-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/mathvariant-auto-selection-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/overflow/computed-value-001.html:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/clipboard-event-handlers.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/html-or-foreign-element-interfaces.tentative-expected.txt:
* LayoutTests/mathml/arbitrary-markup-expected.txt:
* LayoutTests/mathml/fenced-whitespace-separators-crash-expected.txt:
* LayoutTests/mathml/mfrac-nested-expected.txt:
* LayoutTests/mathml/msub-anonymous-child-render-crash-expected.txt:
* LayoutTests/mathml/non-core/mathvariant/attributes-mathvariant-expected.html: Renamed from LayoutTests/mathml/presentation/attributes-mathvariant-expected.html.
* LayoutTests/mathml/non-core/mathvariant/attributes-mathvariant.html: Renamed from LayoutTests/mathml/presentation/attributes-mathvariant.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-basic-transforms-with-default-font.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-fraktur.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-sans-serif.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-script.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-case-sensitivity.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-double-struck-font-style-font-weight.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-double-struck.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-fraktur.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-inheritance-expected.html: Renamed from LayoutTests/mathml/presentation/mathvariant-inheritance-expected.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-inheritance.html: Renamed from LayoutTests/mathml/presentation/mathvariant-inheritance.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-initial.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-looped.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-monospace.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-sans-serif-bold-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-sans-serif-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-sans-serif.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-script.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-stretched.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-tailed.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-tokens-expected.html: Renamed from LayoutTests/mathml/presentation/mathvariant-tokens-expected.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-tokens.html: Renamed from LayoutTests/mathml/presentation/mathvariant-tokens.html.
* LayoutTests/mathml/non-core/mathvariant/tokenElements-mathvariant-expected.html: Renamed from LayoutTests/mathml/presentation/tokenElements-mathvariant-expected.html.
* LayoutTests/mathml/non-core/mathvariant/tokenElements-mathvariant.html: Renamed from LayoutTests/mathml/presentation/tokenElements-mathvariant.html.
* LayoutTests/mathml/presentation/foreign-element-in-annotation-xml-expected.txt:
* LayoutTests/mathml/presentation/stretchy-depth-height-expected.txt:
* LayoutTests/mathml/presentation/sup-nested-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/size-and-position-of-stretchy-fences-with-default-font-001-expected.txt:
* LayoutTests/platform/glib/mathml/presentation/foreign-element-in-annotation-xml-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/size-and-position-of-stretchy-fences-with-default-font-001-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/floats/not-floating-001-expected.txt:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/css/mathmlCoreExtras.css:
* Source/WebCore/css/mathmlCoreMathvariant.css: Copied from Source/WebCore/css/mathmlCoreExtras.css.
(#if defined(ENABLE_MATHML) &amp;&amp; ENABLE_MATHML):
* Source/WebCore/mathml/MathMLElement.cpp:
(WebCore::MathMLElement::hasPresentationalHintsForAttribute const):
(WebCore::MathMLElement::collectPresentationalHintsForAttribute):
* Source/WebCore/mathml/MathMLFractionElement.h:
* Source/WebCore/mathml/MathMLMathElement.cpp:
(WebCore::MathMLMathElement::attributeChanged):
(WebCore::MathMLMathElement::acceptsLegacyMathVariantAttribute):
* Source/WebCore/mathml/MathMLMathElement.h:
* Source/WebCore/mathml/MathMLPresentationElement.cpp:
(WebCore::MathMLPresentationElement::specifiedMathVariant):
(WebCore::MathMLPresentationElement::attributeChanged):
* Source/WebCore/mathml/MathMLPresentationElement.h:
(WebCore::MathMLPresentationElement::acceptsLegacyMathVariantAttribute):
(WebCore::MathMLPresentationElement::acceptsMathVariantAttribute): Deleted.
* Source/WebCore/mathml/MathMLRowElement.cpp:
(WebCore::MathMLRowElement::acceptsLegacyMathVariantAttribute):
(WebCore::MathMLRowElement::acceptsMathVariantAttribute):
* Source/WebCore/mathml/MathMLRowElement.h:
* Source/WebCore/mathml/MathMLScriptsElement.h:
* Source/WebCore/mathml/MathMLTokenElement.cpp:
(WebCore::MathMLTokenElement::acceptsLegacyMathVariantAttribute):
* Source/WebCore/mathml/MathMLTokenElement.h:
* Source/WebCore/rendering/mathml/RenderMathMLToken.cpp:
(WebCore::RenderMathMLToken::computePreferredLogicalWidths):
* Source/WebCore/style/InspectorCSSOMWrappers.cpp:
(WebCore::Style::InspectorCSSOMWrappers::collectDocumentWrappers):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/style/UserAgentStyle.h:

Canonical link: <a href="https://commits.webkit.org/312406@main">https://commits.webkit.org/312406@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d20e7395e689f9fedc91ce0b9bafa6609a58f888

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/157701 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/31038 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/24231 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/166525 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/111783 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/159572 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/31173 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/31040 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/123122 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/85761 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/160659 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/24394 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/141609 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/102776 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/23450 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/21734 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/14296 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/149752 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/133130 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/19426 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/170232 "Built successfully") | 
| [✅ 🛠 🧪 jsc-debug-arm64](https://ews-build.webkit.org/#/builders/171/builds/18536 "Built successfully and passed tests") | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/13567 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/21047 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/131312 "Passed tests") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/30784 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/25804 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/131426 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/30722 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/141220 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/88560 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/24327 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/25186 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/18025 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/189829 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/30274 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/94754 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/48754 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/29795 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/30025 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/29922 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eerii committed:
Align mathvariant attribute with MathML Core
https://bugs.webkit.org/show_bug.cgi?id=202302

Reviewed by NOBODY (OOPS!).

The CoreMathMLDeprecateLegacyMathvariant preference is used to disable the
previous mathvariant behaviour and other variations. The UA stylesheet is
updated to align with the spec, implementing the italic transform on mi
nodes using text-transform: math-auto.

The mathvariant attribute can only be present on mi nodes
(https://w3c.github.io/mathml-core/#dfn-mathvariant). If specified,
it resets the text-transform to none using a presentational hint.

* LayoutTests/accessibility/ios-simulator/math.html:
* LayoutTests/accessibility/mac/math-alttext.html:
* LayoutTests/accessibility/mac/mathml-elements.html:
* LayoutTests/accessibility/mac/mathml-root.html:
* LayoutTests/accessibility/math-fenced.html:
* LayoutTests/accessibility/math-mo-fences-and-separators.html:
* LayoutTests/accessibility/math-multiscript-attributes.html:
* LayoutTests/accessibility/math-text.html:
* LayoutTests/fast/css/readonly-pseudoclass-common-element-expected.txt:
* LayoutTests/fast/css/readwrite-pseudoclass-editable-expected.txt:
* LayoutTests/fast/layers/accumulated-offset-overflow-Render-geometry-map-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/mathml-html-serializtion-on-copy.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/documents/dom-tree-accessors/document.getElementsByName/document.getElementsByName-namespace-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/documents/dom-tree-accessors/document.getElementsByName/document.getElementsByName-namespace-xhtml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/math-parse03-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/direction/direction-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/size-and-position-of-stretchy-fences-with-default-font-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/attribute-mapping-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/floats/not-floating-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/mathvariant-auto-selection-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/overflow/computed-value-001.html:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/clipboard-event-handlers.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/html-or-foreign-element-interfaces.tentative-expected.txt:
* LayoutTests/mathml/arbitrary-markup-expected.txt:
* LayoutTests/mathml/fenced-whitespace-separators-crash-expected.txt:
* LayoutTests/mathml/mfrac-nested-expected.txt:
* LayoutTests/mathml/msub-anonymous-child-render-crash-expected.txt:
* LayoutTests/mathml/non-core/mathvariant/attributes-mathvariant-expected.html: Renamed from LayoutTests/mathml/presentation/attributes-mathvariant-expected.html.
* LayoutTests/mathml/non-core/mathvariant/attributes-mathvariant.html: Renamed from LayoutTests/mathml/presentation/attributes-mathvariant.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-basic-transforms-with-default-font.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-fraktur.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-sans-serif.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-script.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-case-sensitivity.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-double-struck-font-style-font-weight.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-double-struck.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-fraktur.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-inheritance-expected.html: Renamed from LayoutTests/mathml/presentation/mathvariant-inheritance-expected.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-inheritance.html: Renamed from LayoutTests/mathml/presentation/mathvariant-inheritance.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-initial.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-looped.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-monospace.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-sans-serif-bold-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-sans-serif-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-sans-serif.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-script.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-stretched.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-tailed.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-tokens-expected.html: Renamed from LayoutTests/mathml/presentation/mathvariant-tokens-expected.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-tokens.html: Renamed from LayoutTests/mathml/presentation/mathvariant-tokens.html.
* LayoutTests/mathml/non-core/mathvariant/tokenElements-mathvariant-expected.html: Renamed from LayoutTests/mathml/presentation/tokenElements-mathvariant-expected.html.
* LayoutTests/mathml/non-core/mathvariant/tokenElements-mathvariant.html: Renamed from LayoutTests/mathml/presentation/tokenElements-mathvariant.html.
* LayoutTests/mathml/presentation/foreign-element-in-annotation-xml-expected.txt:
* LayoutTests/mathml/presentation/stretchy-depth-height-expected.txt:
* LayoutTests/mathml/presentation/sup-nested-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/size-and-position-of-stretchy-fences-with-default-font-001-expected.txt:
* LayoutTests/platform/glib/mathml/presentation/foreign-element-in-annotation-xml-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/size-and-position-of-stretchy-fences-with-default-font-001-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/floats/not-floating-001-expected.txt:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/css/mathmlCoreExtras.css:
* Source/WebCore/css/mathmlCoreMathvariant.css: Copied from Source/WebCore/css/mathmlCoreExtras.css.
(#if defined(ENABLE_MATHML) && ENABLE_MATHML):
* Source/WebCore/mathml/MathMLElement.cpp:
(WebCore::MathMLElement::hasPresentationalHintsForAttribute const):
(WebCore::MathMLElement::collectPresentationalHintsForAttribute):
* Source/WebCore/mathml/MathMLFractionElement.h:
* Source/WebCore/mathml/MathMLMathElement.cpp:
(WebCore::MathMLMathElement::attributeChanged):
(WebCore::MathMLMathElement::acceptsLegacyMathVariantAttribute):
* Source/WebCore/mathml/MathMLMathElement.h:
* Source/WebCore/mathml/MathMLPresentationElement.cpp:
(WebCore::MathMLPresentationElement::specifiedMathVariant):
(WebCore::MathMLPresentationElement::attributeChanged):
* Source/WebCore/mathml/MathMLPresentationElement.h:
(WebCore::MathMLPresentationElement::acceptsLegacyMathVariantAttribute):
(WebCore::MathMLPresentationElement::acceptsMathVariantAttribute): Deleted.
* Source/WebCore/mathml/MathMLRowElement.cpp:
(WebCore::MathMLRowElement::acceptsLegacyMathVariantAttribute):
(WebCore::MathMLRowElement::acceptsMathVariantAttribute):
* Source/WebCore/mathml/MathMLRowElement.h:
* Source/WebCore/mathml/MathMLScriptsElement.h:
* Source/WebCore/mathml/MathMLTokenElement.cpp:
(WebCore::MathMLTokenElement::acceptsLegacyMathVariantAttribute):
* Source/WebCore/mathml/MathMLTokenElement.h:
* Source/WebCore/rendering/mathml/RenderMathMLToken.cpp:
(WebCore::RenderMathMLToken::computePreferredLogicalWidths):
* Source/WebCore/style/InspectorCSSOMWrappers.cpp:
(WebCore::Style::InspectorCSSOMWrappers::collectDocumentWrappers):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/style/UserAgentStyle.h:

fred-wang commented on Source/WebCore/css/mathmlCoreMathvariant.css:
```diff
@@ -0,0 +1,31 @@
+ */
+
+/* https://w3c.github.io/mathml-core/#user-agent-stylesheet */
+/* This file contains the styles that should be gated by the preference MathMLDisableLegacyMathvariant */
```
nit: sentence should end with a period per https://webkit.org/code-style-guidelines/#comments-sentences

eerii replied:
Sorry about that, I'll also fix the one in `mathmlCoreExtras.css`.

fred-wang commented on Source/WebCore/rendering/mathml/RenderMathMLToken.cpp:
```diff
@@ -76,6 +77,9 @@ void RenderMathMLToken::computePreferredLogicalWidths()
     ASSERT(needsPreferredLogicalWidthsUpdate());
 
+    if (document().settings().cssTextTransformMathAutoEnabled())
+        return RenderMathMLBlock::computePreferredLogicalWidths();
```
I wonder if we can have weird situation with the cached `m_mathVariant*` members when ones dynamically enables/disables the preference, but hopefully current code is ok.

eerii replied:
Maybe. However, I have checked a few different tests enabling and disabling the preference manually and the behaviour seems to be correct and it doesn't have weird artifacts.

fred-wang commented on LayoutTests/platform/mac/accessibility/math-fenced-expected.txt:
```diff
@@ -10,10 +10,10 @@ AXRole: AXGroup AXSubrole: AXDocumentMath AXValue:
         AXRole: AXStaticText AXSubrole: AXMathSeparatorOperator AXValue: ,
         AXRole: AXGroup AXSubrole: AXMathIdentifier AXValue:
-            AXRole: AXStaticText AXSubrole:  AXValue: a
+            AXRole: AXStaticText AXSubrole:  AXValue: �
```
mmh, this seems to be broken?

eerii replied:
This is because of the selection issue, it is only reading the first part of the character (here it is not escaped like in `mathvariant-auto-selection`, it is rendered)

fred-wang commented on LayoutTests/mathml/non-core/scripts-subsup.html:
```diff
@@ -1,4 +1,4 @@
-<!doctype html>
+<!doctype html><!-- webkit-test-runner [ CoreMathMLEnabled=false ] -->
```
I'm not sure what the changes to these subsup and underover tests are about?

maybe  we should rewrite them to use mn and numbers instead of mi and letters, if that cause weird interaction with text-transform.

eerii replied:
I think the issue here is that the test was wrong before (it was 0.71 not 0.75), and the slight tweaks triggered the failure. Since this is also changed in #52066, I think it makes sense to remove the changes from here and wait until `font-size: math` is merged.

fred-wang commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
nit: I guess more explanation would be nice for people who don't have the full context of the MathML changes although it's probably difficult to explain concisely.

eerii replied:
I mentioned the MathML Core spec and that only italic is supported now. It is still missing some more context but it was tricky to fit more into the small description.

eerii committed:
Align mathvariant attribute with MathML Core
https://bugs.webkit.org/show_bug.cgi?id=202302

Reviewed by Frédéric Wang Nélar.

The CoreMathMLDeprecateLegacyMathvariant preference is used to disable the
previous mathvariant behaviour and other variations. The UA stylesheet is
updated to align with the spec, implementing the italic transform on mi
nodes using text-transform: math-auto.

The mathvariant attribute can only be present on mi nodes
(https://w3c.github.io/mathml-core/#dfn-mathvariant). If specified,
it resets the text-transform to none using a presentational hint.

* LayoutTests/accessibility/ios-simulator/math.html:
* LayoutTests/accessibility/mac/math-alttext.html:
* LayoutTests/accessibility/mac/mathml-elements.html:
* LayoutTests/accessibility/mac/mathml-root.html:
* LayoutTests/accessibility/math-fenced.html:
* LayoutTests/accessibility/math-mo-fences-and-separators.html:
* LayoutTests/accessibility/math-multiscript-attributes.html:
* LayoutTests/accessibility/math-text.html:
* LayoutTests/fast/css/readonly-pseudoclass-common-element-expected.txt:
* LayoutTests/fast/css/readwrite-pseudoclass-editable-expected.txt:
* LayoutTests/fast/layers/accumulated-offset-overflow-Render-geometry-map-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/other/mathml-html-serializtion-on-copy.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/documents/dom-tree-accessors/document.getElementsByName/document.getElementsByName-namespace-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/documents/dom-tree-accessors/document.getElementsByName/document.getElementsByName-namespace-xhtml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/math-parse03-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/direction/direction-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/size-and-position-of-stretchy-fences-with-default-font-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/attribute-mapping-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/floats/not-floating-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/mathvariant-auto-selection-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/overflow/computed-value-001.html:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/clipboard-event-handlers.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/html-or-foreign-element-interfaces.tentative-expected.txt:
* LayoutTests/mathml/arbitrary-markup-expected.txt:
* LayoutTests/mathml/fenced-whitespace-separators-crash-expected.txt:
* LayoutTests/mathml/mfrac-nested-expected.txt:
* LayoutTests/mathml/msub-anonymous-child-render-crash-expected.txt:
* LayoutTests/mathml/non-core/mathvariant/attributes-mathvariant-expected.html: Renamed from LayoutTests/mathml/presentation/attributes-mathvariant-expected.html.
* LayoutTests/mathml/non-core/mathvariant/attributes-mathvariant.html: Renamed from LayoutTests/mathml/presentation/attributes-mathvariant.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-basic-transforms-with-default-font.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-fraktur.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-sans-serif.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold-script.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-bold.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-case-sensitivity.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-double-struck-font-style-font-weight.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-double-struck.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-fraktur.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-inheritance-expected.html: Renamed from LayoutTests/mathml/presentation/mathvariant-inheritance-expected.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-inheritance.html: Renamed from LayoutTests/mathml/presentation/mathvariant-inheritance.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-initial.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-looped.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-monospace.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-sans-serif-bold-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-sans-serif-italic.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-sans-serif.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-script.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-stretched.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-tailed.html:
* LayoutTests/mathml/non-core/mathvariant/mathvariant-tokens-expected.html: Renamed from LayoutTests/mathml/presentation/mathvariant-tokens-expected.html.
* LayoutTests/mathml/non-core/mathvariant/mathvariant-tokens.html: Renamed from LayoutTests/mathml/presentation/mathvariant-tokens.html.
* LayoutTests/mathml/non-core/mathvariant/tokenElements-mathvariant-expected.html: Renamed from LayoutTests/mathml/presentation/tokenElements-mathvariant-expected.html.
* LayoutTests/mathml/non-core/mathvariant/tokenElements-mathvariant.html: Renamed from LayoutTests/mathml/presentation/tokenElements-mathvariant.html.
* LayoutTests/mathml/presentation/foreign-element-in-annotation-xml-expected.txt:
* LayoutTests/mathml/presentation/stretchy-depth-height-expected.txt:
* LayoutTests/mathml/presentation/sup-nested-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/size-and-position-of-stretchy-fences-with-default-font-001-expected.txt:
* LayoutTests/platform/glib/mathml/presentation/foreign-element-in-annotation-xml-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/size-and-position-of-stretchy-fences-with-default-font-001-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/floats/not-floating-001-expected.txt:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/css/mathmlCoreExtras.css:
* Source/WebCore/css/mathmlCoreMathvariant.css: Copied from Source/WebCore/css/mathmlCoreExtras.css.
(#if defined(ENABLE_MATHML) && ENABLE_MATHML):
* Source/WebCore/mathml/MathMLElement.cpp:
(WebCore::MathMLElement::hasPresentationalHintsForAttribute const):
(WebCore::MathMLElement::collectPresentationalHintsForAttribute):
* Source/WebCore/mathml/MathMLFractionElement.h:
* Source/WebCore/mathml/MathMLMathElement.cpp:
(WebCore::MathMLMathElement::attributeChanged):
(WebCore::MathMLMathElement::acceptsLegacyMathVariantAttribute):
* Source/WebCore/mathml/MathMLMathElement.h:
* Source/WebCore/mathml/MathMLPresentationElement.cpp:
(WebCore::MathMLPresentationElement::specifiedMathVariant):
(WebCore::MathMLPresentationElement::attributeChanged):
* Source/WebCore/mathml/MathMLPresentationElement.h:
(WebCore::MathMLPresentationElement::acceptsLegacyMathVariantAttribute):
(WebCore::MathMLPresentationElement::acceptsMathVariantAttribute): Deleted.
* Source/WebCore/mathml/MathMLRowElement.cpp:
(WebCore::MathMLRowElement::acceptsLegacyMathVariantAttribute):
(WebCore::MathMLRowElement::acceptsMathVariantAttribute):
* Source/WebCore/mathml/MathMLRowElement.h:
* Source/WebCore/mathml/MathMLScriptsElement.h:
* Source/WebCore/mathml/MathMLTokenElement.cpp:
(WebCore::MathMLTokenElement::acceptsLegacyMathVariantAttribute):
* Source/WebCore/mathml/MathMLTokenElement.h:
* Source/WebCore/rendering/mathml/RenderMathMLToken.cpp:
(WebCore::RenderMathMLToken::computePreferredLogicalWidths):
* Source/WebCore/style/InspectorCSSOMWrappers.cpp:
(WebCore::Style::InspectorCSSOMWrappers::collectDocumentWrappers):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/style/UserAgentStyle.h:

Canonical link: https://commits.webkit.org/312406@main
